cache/demux.go

```diff
@@ -27,7 +27,7 @@ type demux struct {
 	// activeWatchers & laggingWatchers hold the first revision the watcher still needs (nextRev).
 	activeWatchers  map[*watcher]int64
 	laggingWatchers map[*watcher]int64
-	history         *ringBuffer
```

Comment:
FYI, you don't need a `history *ringBuffer`, `history ringBuffer` would be enough and would be cleaner and avoid additional dereference.